### PR TITLE
fix: Change socket timeout default to 0 [PORT]

### PR DESCRIPTION
**Pull request by nbbeeken**
Socket timeout by default is infinity.

NODE-2835

### CODE REVIEW

**Review by reggi — APPROVED** [review]
LGTM

File: test/functional/mongo_client_options.test.js [hunk 1]
```diff
@@ -96,6 +97,17 @@ describe('MongoClient Options', function() {
...
+    const client = new MongoClient(this.configuration.url());
+    client.connect(() => {
+      expect(client.s.options.socketTimeoutMS).to.deep.equal(0);
+      for (const connection of client.topology.s.coreTopology.connections()) {
+        expect(connection.socketTimeout).to.deep.equal(0);
+      }
+      done();
+    });
+  });
+
```
Comment (nbbeeken) [inline test/functional/mongo_client_options.test.js:110]: @mbroadst I saw your comment about missing testing on the 4.0 version of this, added a test here that I'll also add to 4.0 if this is all good ✅

Comment (mbroadst) [reply test/functional/mongo_client_options.test.js:110]: LGTM!

**Review by mbroadst — APPROVED** [review]

**Review by emadum — APPROVED** [review]

File: test/tools/runner/index.js [hunk 2]
```diff
@@ -86,7 +86,7 @@ before(function(_done) {
         }
 
         this.configuration = new TestConfiguration(parsedURI, context);
-        done();
+        client.close(done);
```
Comment (mbroadst) [inline test/tools/runner/index.js:89]: I've fallen for this before myself :) We _do_ call `close` inside of `done` on L65

File: test/functional/operation_promises_example.test.js [hunk 3]
```diff
@@ -3149,22 +3149,28 @@ describe('Operation (Promises)', function() {
...
-        // LINE   test = require('assert');
-        // LINE const client = new MongoClient('mongodb://localhost:27017/test');
-        // LINE client.connect().then(() => {
-        // LINE   var db = client.db('test);
-        // REPLACE configuration.writeConcernMax() WITH {w:1}
-        // REMOVE-LINE done();
-        // BEGIN
-        // Retry to get the collection, should work as it's now created
-        return db.collections().then(function(collections) {
+      return client
```
Comment (mbroadst) [inline test/functional/operation_promises_example.test.js:3152]: what's motivating this test change?

File: test/functional/operation_generators_example.test.js [hunk 4]
```diff
@@ -2931,6 +2931,7 @@ describe('Operation (Generators)', function() {
           .newClient(configuration.writeConcernMax(), { poolSize: 1 })
           .connect();
         var db = client.db(configuration.db);
+        yield db.createCollection('example');
```
Comment (mbroadst) [inline test/functional/operation_generators_example.test.js:2934]: what's motivating this change?

File: test/functional/collection.test.js [hunk 5]
```diff
@@ -799,7 +799,7 @@ describe('Collection', function() {
 
           db.listCollections().toArray((err, documents) => {
             expect(err).to.not.exist;
-            expect(documents.length > 1).to.be.true;
+            expect(documents.length >= 1).to.be.true;
```
Comment (mbroadst) [inline test/functional/collection.test.js:802]: what changed here?

**Review by mbroadst — CHANGES_REQUESTED** [review]

File: test/functional/collection.test.js (hunk 5 above)
Comment (nbbeeken) [reply test/functional/collection.test.js:802]: This test fails if you run it in isolation, it depends upon other tests to create multiple collections when it only creates one itself.

File: test/functional/operation_generators_example.test.js (hunk 4 above)
Comment (nbbeeken) [reply test/functional/operation_generators_example.test.js:2934]: This test fails if you run it in isolation, it depends upon other tests to create collections.

File: test/functional/operation_promises_example.test.js (hunk 3 above)
Comment (nbbeeken) [reply test/functional/operation_promises_example.test.js:3152]: This test fails if you run it in isolation, it depends upon other tests to create the collection.

File: test/tools/runner/index.js (hunk 2 above)
Comment (nbbeeken) [reply test/tools/runner/index.js]: Whoops! thanks.

**Review by mbroadst — APPROVED** [review]
LGTM

**Pull request merged**